lib/cisco_node_utils/span_session.rb

```diff
@@ -0,0 +1,123 @@
+# Insert appropriate license here
+require_relative 'node_util'
+require_relative 'interface'
+
+module Cisco
+  # node utils class for SPAN (switchport analyzer) sessions
+  class SpanSession < NodeUtil
+    attr_reader :session_id
+
+    def initialize(session_id, instantiate=true)
+      validate_args(session_id.to_i)
+      create if instantiate
+    end
+
+    def self.sessions
+      hash = {}
+      all = config_get('span_session', 'all_sessions')
+      return hash if all.nil?
+
+      all.each do |id|
+        hash[id] = SpanSession.new(id, false)
+      end
+      hash
+    end
+
+    def validate_args(session_id)
+      fail TypeError unless session_id.is_a?(Integer)
+      @session_id = session_id
+    end
+
+    def create
+      config_set('span_session', 'create', id: @session_id)
+    end
+
+    def destroy
+      config_set('span_session', 'destroy', id: @session_id)
+    end
+
+    def description
+      config_get('span_session', 'description', id: @session_id)
+    end
+
+    def description=(val)
+      val = val.to_s
+      if val.empty?
+        config_set('span_session', 'description', id: @session_id,
+                    state: 'no', description: '')
+      else
+        config_set('span_session', 'description', id: @session_id,
+                    state: '', description: val)
+      end
+    end
+
+    def destination
+      config_get('span_session', 'destination', id: @session_id)
+    end
+
+    def destination=(int)
+      # fail if int is not a valid interface
+      fail TypeError unless Interface.interfaces.key?(int)
```

Comment:
You need to downcase `int` so that you find it in the list of available interfaces `...(int.downcase)`